### PR TITLE
NET-6821 Disable Terminating Gateway Auto Host Header Rewrite

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
- Exposing configuration to allow users of the terminating gateway to turn off host header rewrites at the service level 

### Testing & Reproduction steps
- build local consul 
- apply the following terminating gateway 
```
Kind = "terminating-gateway"
Name = "test-gateway"
Services = [
 {
   Name = "default-behavior"
 },
 {
   Name = "default-behavior"
   DisableAutoHostRewrite = true
 }
]

```

Observe the generated XDS and note that the default behavior is to auto host rewrite, but not when DisableAutoHostRewrite is set to true.

### PR Checklist

* [ X ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: agent/consul/state/catalog.go [hunk 1]
```diff
@@ -3757,14 +3757,15 @@ func terminatingConfigGatewayServices(
...
-			ServiceKind: kind,
+			Gateway:         gateway,
+			Service:         structs.NewServiceName(svc.Name, &svc.EnterpriseMeta),
+			GatewayKind:     structs.ServiceKindTerminatingGateway,
+			KeyFile:         svc.KeyFile,
+			CertFile:        svc.CertFile,
+			CAFile:          svc.CAFile,
+			SNI:             svc.SNI,
+			ServiceKind:     kind,
+			AutoHostRewrite: !svc.DisableAutoHostRewrite,
```
Comment (sarahalsmiller) [inline agent/consul/state/catalog.go:3768]: My logic here is that by making this a flag you have to set to true it won't change any existing behavior in existing terminating gateway setups.

**Comment by david-yu** [comment]
@sarahalsmiller Will an accompanying PR to update CRDs also follow? Looking forward this PR being merged!

**Comment by sarahalsmiller** [comment]
> @sarahalsmiller Will an accompanying PR to update CRDs also follow? Looking forward this PR being merged!

Yes it will. In my experience its just easier to wait until the original PR merges into core so you don't accidentally point the go.mod file at a commit that doesn't exist in main.

File: website/content/docs/connect/config-entries/terminating-gateway.mdx [hunk 2]
```diff
@@ -679,6 +679,12 @@ spec:
           `An optional hostname or domain name to specify during the TLS handshake. This option will also configure [strict SAN matching](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-field-extensions-transport-sockets-tls-v3-certificatevalidationcontext-match-typed-subject-alt-names), which requires
             the external services to have certificates with SANs, not having which will result in \`CERTIFICATE_VERIFY_FAILED\` error.`,
         },
+        {
+          name: 'DisableAutoHostRewrite',
+          type: 'bool: ""',
+          description:
+            'When set to true, Terminating Gateway will not modify the incoming requests host header for this service.',
```
Comment (sarahalsmiller) [inline website/content/docs/connect/config-entries/terminating-gateway.mdx:686]:
```suggestion
           'When set to true, disables the terminating gateway's ability to modify this service's incoming requests host header.',
```

**Review by missylbytes — APPROVED** [review]
LGTM when all tests pass

**Pull request merged**